Detectors/FIT/simulation/src/Digitizer.cxx

```diff
@@ -10,54 +10,234 @@
+  {
+    if(ch_signal_nPe[ch_iter] != 0)
+    {
+      ch_signal_MIP[ch_iter] = ch_signal_nPe[ch_iter] / nPe_in_mip;
+      ch_signal_time[ch_iter] = ch_signal_time[ch_iter] / (float)ch_signal_nPe[ch_iter];
+     } 
+
+    //if ampl less than cfd trh adc and cfd has no signal
+    if(ch_signal_MIP[ch_iter] < CFD_trsh_mip)
+    {
+	ch_signal_MIP[ch_iter] = 0.;
+	ch_signal_time[ch_iter] = 0.;	
     }
-  } // end of loop over hits
+  }
+ 
+  // Calculating triggers -----------------------------------------------------
+  Int_t n_hit_A = 0., n_hit_C = 0.;
+  Double_t mean_time_A = 0.;
+  Double_t mean_time_C = 0.;
+  Double_t summ_ampl_A = 0.;
+  Double_t summ_ampl_C = 0.;
+  Double_t vertex_time;
+
+  Bool_t Is_A, Is_C;
+  Bool_t Is_Central;
+  Bool_t Is_SemiCentral;
+  Bool_t Is_Vertex;
+
+  for (Int_t ch_iter = 0; ch_iter < nMCPs; ch_iter++)
+  {
+    Bool_t is_A_side = (ch_iter <= 4 * Geometry::NCellsA);
+    Bool_t is_time_in_trg_gate = (ch_signal_time[ch_iter] > BC_clk_center-time_trg_gate*0.5)
+                               &&(ch_signal_time[ch_iter] < BC_clk_center+time_trg_gate*0.5);
+    //  if (ch_signal_time[ch_iter]>0)
+    if (ch_signal_MIP[ch_iter] == 0.)
+      continue;
+    if(ch_signal_MIP[ch_iter] <CFD_trsh_mip) continue;
+    if(!is_time_in_trg_gate) continue;
 
-  for (Int_t ipmt = 0; ipmt < nMCPs; ipmt++) {
-    if (amp[ipmt] > mAmpThreshold) {
-      cfd[ipmt] = cfd[ipmt] / Float_t(amp[ipmt]); //mean time on 1 quadrant
-      cfd[ipmt] = (gRandom->Gaus(cfd[ipmt], 50)) / Geometry::ChannelWidth;
-      mDigits->emplace_back(timeframe, ipmt, cfd[ipmt], Float_t(amp[ipmt]), bc);
+    if(is_A_side)
+    {
+	n_hit_A++;
+	summ_ampl_A += ch_signal_MIP[ch_iter];
+	mean_time_A += ch_signal_time[ch_iter];
     }
-  } // end of loop over PMT
+    else
+    {
+	n_hit_C++;
+	summ_ampl_C += ch_signal_MIP[ch_iter];
+	mean_time_C += ch_signal_time[ch_iter];
+    }
+
+  }
+
+  Is_A = n_hit_A > 0;
+  Is_C = n_hit_C > 0;
+  Is_Central = summ_ampl_A + summ_ampl_C >= trg_central_trh;
+  Is_SemiCentral = summ_ampl_A + summ_ampl_C >= trg_semicentral_trh;
+
+  mean_time_A = Is_A? mean_time_A/n_hit_A : 0.;
+  mean_time_C = Is_C? mean_time_C/n_hit_C : 0.;
+  vertex_time = (mean_time_A + mean_time_C)*.5 ;
+  Is_Vertex = (vertex_time > trg_vertex_min) && (vertex_time < trg_vertex_max);
+  // --------------------------------------------------------------------------
+
+
+  //fillig digit
+  digit->setTime(digit_timeframe);
+  digit->setBC(mEventID);
+  digit->setTriggers(Is_A, Is_C, Is_Central, Is_SemiCentral, Is_Vertex);
+
+  std::vector<ChannelData> mChDgDataArr;
+  //  LOG(DEBUG) << "nMCP : :IsA : hit nPe : hit mTime : sig MIP : sig mTime" << FairLogger::endl;
+  for (Int_t ch_iter = 0; ch_iter < nMCPs; ch_iter++) {
+    if (ch_signal_MIP[ch_iter] > CFD_trsh_mip) {
+      Float_t smeared_time = gRandom->Gaus(ch_signal_time[ch_iter], 0.050) + BCEventTime;
+      mChDgDataArr.emplace_back(ChannelData{ ch_iter, smeared_time, ch_signal_MIP[ch_iter]} );
```

Comment:
the CI is reporting a compilation error here due to a narrowing conversion of double to float in ch_signal_MIP.
So either a cast needs to be applied or store ch_signal_MIP in float as well.

Moreover, if ChannelData gets a constructor this line can be simplified to

```
mChDgDataArr.emplace_back( ch_iter, smeared_time, ch_signal_MIP[ch_iter] );
```